module3/homework/passwordCheck/validation.hpp

```diff
@@ -1,2 +1,21 @@
-// TODO: I'm empty :) Put enum and function headers here.
-// Don't forget the header guard
\ No newline at end of file
+#pragma once
+
+#include <string>
+
+enum class ErrorCode
+{
+    Ok,
+    PasswordNeedsAtLeastNineCharacters,
+    PasswordNeedsAtLeastOneNumber,
+    PasswordNeedsAtLeastOneSpecialCharacter,
+    PasswordNeedsAtLeastOneUppercaseLetter,
+    PasswordsDoesNotMatch
+};
+
+std::string getErrorMessage(const ErrorCode &err);
+
+bool doesPasswordsMatch(const std::string &passwd1, const std::string &passwd2);
+
+ErrorCode checkPasswordRules(const std::string &passwd);
+
+ErrorCode checkPassword(const std::string &passwd1, const std::string &passwd2);
```

Comment:
Enums are stored as ints, so I think that you don't need to use reference. Especially that you don't want to modify it.
```suggestion
std::string getErrorMessage(ErrorCode err);

bool doesPasswordsMatch(const std::string& passwd1, const std::string& passwd2);

ErrorCode checkPasswordRules(const std::string& passwd);

ErrorCode checkPassword(const std::string& passwd1, const std::string& passwd2);
```

Reply:
@nauka-programowania-MA Please comment if `enum class` should be pass by `const &` in this case. Usually there is no need to pass POD types via `&` but `enum class` is more complex type. Compiler probably optimize it so there won't be any difference in the machine code.